plugins/secrets/update.js

```diff
@@ -0,0 +1,56 @@
+/* eslint no-param-reassign: ["error", { "props": false }] */
+'use strict';
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const idSchema = joi.reach(schema.models.secret.base, 'id');
+
+module.exports = () => ({
+    method: 'PUT',
+    path: '/secrets/{id}',
+    config: {
+        description: 'Update a secret',
+        notes: 'Update a specific secret',
+        tags: ['api', 'secrets'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['user']
+        },
+        handler: (request, reply) => {
+            const factory = request.server.app.secretFactory;
+            const credentials = request.auth.credentials;
+            const canAccess = request.server.plugins.secrets.canAccess;
+
+            return factory.get(request.params.id)
+                .then(secret => {
+                    if (!secret) {
+                        throw boom.notFound('Secret does not exist');
+                    }
+
+                    // Make sure that user/build has permission before updating
+                    return canAccess(credentials, secret)
+                        .then(() => {
```

Comment:
Hmmm I think admin access for create, update and remove. write access for get. We can modify canAccess to do this. 
If we want to simplify we can make everything requires admin access.